tests/hw6/test_oop_2.py

```diff
@@ -0,0 +1,129 @@
+from datetime import timedelta
+
+import pytest
+
+from hw6.oop_2 import DeadlineError, Homework, HomeworkResult, Student, Teacher
+
+
+@pytest.fixture(scope="module")
+def teacher_instance():
+    instance = Teacher("Daniil", "Shadrin")
+    yield instance
+
+
+@pytest.fixture(scope="module")
+def good_student_instance():
+    instance = Student("Roman", "Petrov")
+    yield instance
+
+
+@pytest.fixture(scope="module")
+def lazy_student_instance():
+    instance = Student("Lev", "Sokolov")
+    yield instance
+
+
+def test_creating_instances(
+    teacher_instance, good_student_instance, lazy_student_instance
+):
+    teacher = teacher_instance
+    good_student = good_student_instance
+    lazy_student = lazy_student_instance
+
+    the_deadline_was_yesterday = -1
+    homework = Homework(
+        "Text of the homework", timedelta(days=the_deadline_was_yesterday)
+    )
+    assert teacher.first_name == "Daniil"
+    assert teacher.last_name == "Shadrin"
+
+    assert good_student.first_name == "Roman"
+    assert good_student.last_name == "Petrov"
+
+    assert lazy_student.first_name == "Lev"
+    assert lazy_student.last_name == "Sokolov"
+
+    assert homework.is_active() is False
+
+
+def test_check_hw_exception(good_student_instance):
+    with pytest.raises(ValueError):
+        HomeworkResult(good_student_instance, "fff", "Solution", 0)
+
+
+def test_positive_hw_check(teacher_instance, good_student_instance):
+    oop_teacher = Teacher("Daniil", "Shadrin")
```

Comment:
there is teacher_instance parameter here. what is the purpose of recreating it?

Reply:
Done